runtime/util/search.cpp

```diff
@@ -47,19 +47,25 @@ std::unordered_set<block *, HashBlock, KEq> take_search_steps(int64_t depth, blo
   }
 
   std::unordered_set<block *, HashBlock, KEq> results;
+  std::unordered_set<block *, HashBlock, KEq> states_set;
   states.clear();
+  states_set.insert(subject);
   states.push_back(subject);
   while(!states.empty() && depth != 0) {
     state = states.front();
     states.pop_front();
+    states_set.erase(state);
```

Comment:
Maybe I am wrong here, but why couldn't  we just not erase states from this set. If I understand correctly, the way it's implemented right now, we avoid exploring from the same state if it is already in the queue for future exploration. If we never erase states from the `states_set` data structure, we will also be avoiding re-exploring states that have been inserted and in the queue in the past, thus have already been explored.

Reply:
That is a good point.

Reply:
We chose not to do this because it causes a memory leak.